pkg/proxy/proxy.go

```diff
@@ -136,23 +136,44 @@ func generateURL(w http.ResponseWriter, req *http.Request, dport uint16) (*url.U
 		return nil, fmt.Errorf("unable to parse port string: %s", err)
 	}
 
-	key := &Proxy4Key{
+	if pIP.To4() != nil {
+		key := &Proxy4Key{
+			SPort:   uint16(sport),
+			DPort:   dport,
+			Nexthdr: 6,
```

Comment:
Next header fields is the same for both ipv6 and ipv4. The next header for TCP is 6. 4 would be IPinIP I believe.